### PR TITLE
Feature/inf instagan

### PR DESCRIPTION
**Pull request by gcosne**
Comet ML update with image overlay & fake img on validation set every 10 epochs.

### CODE REVIEW

File: instagan/train.py [hunk 1]
```diff
@@ -5,15 +5,38 @@
 from data import CreateDataLoader
 from models import create_model
 from util.visualizer import Visualizer
+from util.visualizer import save_val_set
+from util.visualizer import overlay_flood_mask
+from util.visualizer import fake_img
+import os
 
-experiment = Experiment(project_name="instagan", workspace="vict0rsch")
+comet_exp = None
```
Comment (vict0rsch) [inline instagan/train.py]: why this line?

File: instagan/train.py [hunk 2]
```diff
@@ -58,17 +81,52 @@
...
+        
         print(
             "End of epoch %d / %d \t Time Taken: %d sec"
             % (epoch, opt.niter + opt.niter_decay, time.time() - epoch_start_time)
         )
-        model.update_learning_rate()
+        
+        # INFERENCE CODE 
+        try:os.mkdir(opt_test.results_dir+'epoch'+str(epoch)+'/')
+        except:pass
```
Comment (vict0rsch) [inline instagan/train.py]: Exceptions should be handled. I expect you expect only sir already exists but other stuff may come up, at least print

File: instagan/util/visualizer.py [hunk 3]
```diff
@@ -6,13 +6,84 @@
...
+    name = os.path.splitext(short_path)[0]
+    for label, im_data in visuals.items():
+        im = util.tensor2im(im_data)
+        image_name = '%s_%s.png' % (name, label)
+        save_path = os.path.join(results_dir, image_name)
+        h, w, _ = im.shape
+        if aspect_ratio > 1.0:
+            im = imresize(im, (h, int(w * aspect_ratio)), interp='bicubic')
+        if aspect_ratio < 1.0:
+            im = imresize(im, (int(h / aspect_ratio), w), interp='bicubic')
```
Comment (vict0rsch) [inline instagan/util/visualizer.py]: don't we always want `w * aspect_ratio` ? this formulation does not allow for shrinking

File: instagan/train.py (hunk 1 above)
Comment (gcosne) [reply instagan/train.py]: Correct, I will remove this one

File: instagan/train.py (hunk 2 above)
Comment (gcosne) [reply instagan/train.py]: Exception handled with new commt

File: instagan/util/visualizer.py (hunk 3 above)
Comment (gcosne) [reply instagan/util/visualizer.py]: Default parameter is aspect_ratio of 1, basically it's a modified version of this function saving images and sending them to the web server. (Same file : visualizer.py)
 ```
def save_images(webpage, visuals, image_path, aspect_ratio=1.0, width=256):
    image_dir = webpage.get_image_dir()
    short_path = ntpath.basename(image_path[0])
    name = os.path.splitext(short_path)[0]
    webpage.add_header(name)
    ims, txts, links = [], [], []
    for label, im_data in visuals.items():
        im = util.tensor2im(im_data)
        image_name = '%s_%s.png' % (name, label)
        save_path = os.path.join(image_dir, image_name)
        h, w, _ = im.shape
        if aspect_ratio > 1.0:
            im = imresize(im, (h, int(w * aspect_ratio)), interp='bicubic')
        if aspect_ratio < 1.0:
            im = imresize(im, (int(h / aspect_ratio), w), interp='bicubic')
        util.save_image(im, save_path)

        ims.append(image_name)
        txts.append(label)
        links.append(image_name)
    webpage.add_images(ims, txts, links, width=width)
```
